### PR TITLE
Update the Key Vault index pages on docs.microsoft.com for python

### PR DESCRIPTION
**Pull request by v-xuto**
Update the Key Vault index pages on docs.microsoft.com for python with reference to the link
https://docs.microsoft.com/en-us/dotnet/api/overview/azure/storage?view=azure-dotnet

### CODE REVIEW

File: docs-ref-services/key-vault.md [hunk 1]
```diff
@@ -12,141 +12,28 @@ ms.service: key-vault
...
-pip install azure-keyvault-keys
-pip install azure-keyvault-certificates
-pip install azure-identity
-```
+Use the following version 4.x libraries to work with certificates, keys, and secrets:
 
-## Examples
+| Library | Reference | Package | Source |
+|----------------------------------------|-------------------------------------------------------------|-----------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------|
+|    azure-keyvault-certificates   |    [Reference](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-keyvault-certificates/4.0.0/azure.keyvault.certificates.html)    |    [Pypi](https://pypi.org/project/azure-keyvault-certificates/)    |    [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-keyvault-certificates)    |
```
Comment (heaths) [inline docs-ref-services/key-vault.md:29]: Should check if there's a better reference doc location. We want to retire these locations, but may not have a better location for now. /cc @chlowell

Comment (chlowell) [reply docs-ref-services/key-vault.md:29]: We don't have a better location yet. The preferred doc site doesn't document keyword arguments.

File: docs-ref-services/key-vault.md [hunk 2]
```diff
@@ -12,141 +12,28 @@ ms.service: key-vault
...
+|    azure-keyvault-keys    |    [Reference](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-keyvault-keys/4.0.0/azure.keyvault.keys.html)    |    [Pypi](https://pypi.org/project/azure-keyvault-keys/)    |    [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-keyvault-keys)    |
+|    azure-keyvault-secrets    |    [Reference](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-keyvault-secrets/4.0.0/azure.keyvault.secrets.html)    |    [Pypi](https://pypi.org/project/azure-keyvault-secrets/)    |    [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-keyvault-secrets)    |
```
Comment (chlowell) [inline docs-ref-services/key-vault.md:31]: The latest version of these packages is 4.1.0:
```suggestion
|    azure-keyvault-certificates   |    [Reference](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-keyvault-certificates/4.1.0/azure.keyvault.certificates.html)    |    [Pypi](https://pypi.org/project/azure-keyvault-certificates/)    |    [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-keyvault-certificates)    |
|    azure-keyvault-keys    |    [Reference](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-keyvault-keys/4.1.0/azure.keyvault.keys.html)    |    [Pypi](https://pypi.org/project/azure-keyvault-keys/)    |    [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-keyvault-keys)    |
|    azure-keyvault-secrets    |    [Reference](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-keyvault-secrets/4.1.0/azure.keyvault.secrets.html)    |    [Pypi](https://pypi.org/project/azure-keyvault-secrets/)    |    [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-keyvault-secrets)    |
```

File: docs-ref-services/key-vault.md [hunk 3]
```diff
@@ -12,141 +12,28 @@ ms.service: key-vault
...
+|    azure-keyvault-certificates   |    [Reference](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-keyvault-certificates/4.1.0/azure.keyvault.certificates.html)    |    [Pypi](https://pypi.org/project/azure-keyvault-certificates/)    |    [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-keyvault-certificates)    |
+|    azure-keyvault-keys    |    [Reference](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-keyvault-keys/4.1.0/azure.keyvault.keys.html)    |    [Pypi](https://pypi.org/project/azure-keyvault-keys/)    |    [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-keyvault-keys)    |
+|    azure-keyvault-secrets    |    [Reference](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-keyvault-secrets/4.1.0/azure.keyvault.secrets.html)    |    [Pypi](https://pypi.org/project/azure-keyvault-secrets/)    |    [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-keyvault-secrets)    |
```
Comment (chlowell) [inline docs-ref-services/key-vault.md:31]:
```suggestion
|    azure-keyvault-certificates   |    [Reference](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-keyvault-certificates/4.1.0/azure.keyvault.certificates.html)    |    [PyPI](https://pypi.org/project/azure-keyvault-certificates/)    |    [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-keyvault-certificates)    |
|    azure-keyvault-keys    |    [Reference](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-keyvault-keys/4.1.0/azure.keyvault.keys.html)    |    [PyPI](https://pypi.org/project/azure-keyvault-keys/)    |    [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-keyvault-keys)    |
|    azure-keyvault-secrets    |    [Reference](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-keyvault-secrets/4.1.0/azure.keyvault.secrets.html)    |    [PyPI](https://pypi.org/project/azure-keyvault-secrets/)    |    [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-keyvault-secrets)    |
```

File: docs-ref-services/key-vault.md [hunk 4]
```diff
@@ -12,141 +12,28 @@ ms.service: key-vault
...
-* [Manage Azure Key Vaults][1] 
-* [Azure Key Vault recovery][2]
-
-[1]: https://azure.microsoft.com/resources/samples/key-vault-python-manage/
-[2]: https://azure.microsoft.com/resources/samples/key-vault-recovery-python/
-
-View the [complete list](https://azure.microsoft.com/resources/samples/?platform=python&term=key+vault) of Azure Key Vault samples. 
+|    Library    |    Reference    |    Package    |    Source    |
+|------------------------------------------|-------------------------------------------------------------------|-----------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------|
+|    azure-mgmt-keyvault    |    [Reference](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-mgmt-keyvault/2.2.0/azure.mgmt.keyvault.html)    |    [Pypi](https://pypi.org/project/azure-mgmt-keyvault/)    |    [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-mgmt-keyvault)    |
```
Comment (chlowell) [inline docs-ref-services/key-vault.md:39]:
```suggestion
|    azure-mgmt-keyvault    |    [Reference](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-mgmt-keyvault/2.2.0/azure.mgmt.keyvault.html)    |    [PyPI](https://pypi.org/project/azure-mgmt-keyvault/)    |    [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-mgmt-keyvault)    |
```

**Comment by tzhanl** [comment]
@maggiepint @scbedd  Please review.

**Comment by maggiepint** [comment]
Are we able to add links to the old version to this as well? In it's own table?

All of the old version docs should exist under the legacy moniker

**Comment by v-xuto** [comment]
Hi @maggiepint , 
We can't find the source code and reference for the old version of keyvault, we can only find its package location in PyPI. Do you have any ideas or suggestions?

**Comment by maggiepint** [comment]
The reference is under the 'Legacy' moniker in the ref: https://docs.microsoft.com/en-us/python/api/overview/azure/keyvault?view=azure-python-previous

Getting the links to work and keep the azure-python-previous in the query string is a little bit tricky. I just updated the master branch with Storage index working exactly like I"m asking for this to work. Check there to understand.

**Comment by maggiepint** [comment]
Oh, don't worry about the source in this case. You can just link to the current source - I believe Key Vault is simply an older hash in the same source tree.

File: docs-ref-services/key-vault-index.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+|    azure-keyvault-keys    |    [Reference](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-keyvault-keys/4.1.0/azure.keyvault.keys.html)    |    [PyPI](https://pypi.org/project/azure-keyvault-keys/)    |    [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-keyvault-keys)    |
+|    azure-keyvault-secrets    |    [Reference](https://azuresdkdocs.blob.core.windows.net/$web/python/azure-keyvault-secrets/4.1.0/azure.keyvault.secrets.html)    |    [PyPI](https://pypi.org/project/azure-keyvault-secrets/)    |    [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault/azure-keyvault-secrets)    |
+
+### Version 1.x.x
+
+The source code for the Azure Key Vault client libraries for Python is available on [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/keyvault).
```
Comment (chlowell) [inline docs-ref-services/key-vault-index.md:37]: `azure-keyvault` isn't in the master branch anymore. This is the code of its last release:
```suggestion
The source code for the Azure Key Vault client libraries for Python is available on [GitHub](https://github.com/Azure/azure-sdk-for-python/tree/azure-keyvault_1.1.0/azure-keyvault).
```

**Pull request merged**